### DISCUSSION TITLE
Integrating the model with a microprocessor

### DISCUSSION THREAD

**Discussion by greycellz** [dscripka/openWakeWord#184] [Q&A]
Hi All, 

I have already built out and tested a custom model that works well for us. Do any of you have any sample code that you've used to capture audio and pass that on to the model in the microprocessor device? I've never worked on the microprocessor side of things to apologize in advance if I'm asking something very basic. Any help will be truly appreciated.

Cheers!

**Comment by dscripka** [comment]
I'm not sure what you mean by microprocessor device, but if you are referring to small embedded systems like an ESP32 or a Raspberry Pi Pico, unfortunately the openWakeWord models are too large to run on such small devices. Other projects like [microWakeWord](https://github.com/kahrendt/microWakeWord) are likely a better fit for those devices.
